### PR TITLE
refactor: Use the elvis operator + fix: Handle null checks with the ?? operator

### PR DESCRIPTION
**Pull request by ChristophWurst**
* Resolves: # 

## Summary

Makes code easier to read.

## TODO

- [x] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by st3iny — APPROVED** [review]
:broom:

**Review by miaulalala — APPROVED** [review]

**Comment by miaulalala** [comment]
tenor

File: apps/files_sharing/lib/External/Manager.php [hunk 1]
```diff
@@ -119,7 +119,7 @@
 	 * @throws \Doctrine\DBAL\Exception
 	 */
 	public function addShare($remote, $token, $password, $name, $owner, $shareType, $accepted = false, $user = null, $remoteId = '', $parent = -1) {
-		$user = $user ? $user : $this->uid;
+		$user = $user ?: $this->uid;
```
[the remark this answers is not in the document]
Comment (kesselb) [reply apps/files_sharing/lib/External/Manager.php:122]: https://3v4l.org/b8Yjj

[the remark this answers is not in the document]
Comment (kesselb) [reply apps/files_sharing/lib/External/Manager.php:122]: Would recommend `??`  here.

File: lib/private/User/Database.php [hunk 2]
```diff
@@ -64,7 +64,7 @@ class Database extends ABackend implements
 	public function __construct($eventDispatcher = null, $table = 'users') {
 		$this->cache = new CappedMemoryCache();
 		$this->table = $table;
-		$this->eventDispatcher = $eventDispatcher ? $eventDispatcher : \OCP\Server::get(IEventDispatcher::class);
+		$this->eventDispatcher = $eventDispatcher ?: \OCP\Server::get(IEventDispatcher::class);
```
Comment (kesselb) [inline lib/private/User/Database.php:67]: I think that's also a ?? case.

File: lib/private/Log/LogDetails.php [hunk 3]
```diff
@@ -37,7 +37,7 @@ public function logDetails(string $app, $message, int $level): array {
 		$url = ($request->getRequestUri() !== '') ? $request->getRequestUri() : '--';
 		$method = is_string($request->getMethod()) ? $request->getMethod() : '--';
 		if ($this->config->getValue('installed', false)) {
-			$user = \OC_User::getUser() ? \OC_User::getUser() : '--';
+			$user = \OC_User::getUser() ?: '--';
```
Comment (kesselb) [inline lib/private/Log/LogDetails.php:40]: Uid "0" would also evaluate to '--'

File: lib/private/User/Database.php (hunk 2 above)
Comment (ChristophWurst) [reply lib/private/User/Database.php:67]: How?

File: lib/private/Log/LogDetails.php (hunk 3 above)
Comment (ChristophWurst) [reply lib/private/Log/LogDetails.php:40]: Yes, same as with the ternary operator

File: lib/private/User/Database.php (hunk 2 above)
Comment (kesselb) [reply lib/private/User/Database.php:67]: We know it's either null or EventDispatcher.

**Review by skjnldsv — APPROVED** [review]

**Review by fsamapoor — APPROVED** [review]

File: lib/private/Log/LogDetails.php (hunk 3 above)
Comment (ChristophWurst) [reply lib/private/Log/LogDetails.php]: Fixed :)

**Pull request merged**